### PR TITLE
Save states

### PR DESCRIPTION
**Pull request by ShaunYearStrong**
picture 34

### CODE REVIEW

File: public/video-ui/src/actions/VideoActions/revertAsset.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ function recieveRevertAsset(video) {
 function errorRevertAsset(error) {
   return {
     type:       'SHOW_ERROR',
-    message:    'Could not create video',
+    message:    'Could revert the asset',
```
Comment (akash1810) [inline public/video-ui/src/actions/VideoActions/revertAsset.js]: missing "not"?

File: public/video-ui/src/components/VideoAssets/VideoAssetItem.js [hunk 2]
```diff
@@ -11,6 +11,15 @@ export default class VideoAssetItem extends React.Component {
 
   revertAsset = () => {
     this.props.revertAsset(this.props.asset.version);
+
+    // Update the activeAsset
+    let newData = Object.assign({}, this.props.video.data, {
```
Comment (akash1810) [inline public/video-ui/src/components/VideoAssets/VideoAssetItem.js]: we can `const` rather than `let` here

**Comment by ShaunYearStrong** [comment]
@akash1810 - Made those changes

File: public/video-ui/src/reducers/saveStateReducer.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      return saveStateVals.inprogress;
+
+    case 'ASSET_REVERT_REQUEST':
+      return saveStateVals.inprogress;
+
+    case 'VIDEO_SAVE_REQUEST':
+      return saveStateVals.inprogress;
+
+    case 'VIDEO_GET_RECEIVE':
+      return '';
```
Comment (akash1810) [inline public/video-ui/src/reducers/saveStateReducer.js]: why do we need to do this?

File: public/video-ui/src/components/Video/Create.js [hunk 4]
```diff
@@ -5,10 +5,6 @@ import {blankVideoData} from '../../constants/blankVideoData';
 
 class VideoCreate extends React.Component {
 
-  constructor(props) {
-    super(props);
```
Comment (akash1810) [inline public/video-ui/src/components/Video/Create.js]: how are the props added to this component now?

Comment (ShaunYearStrong) [inline public/video-ui/src/components/Video/Create.js]: On Line 42 where Redux maps the state to the props.

File: public/video-ui/src/reducers/saveStateReducer.js (hunk 3 above)
Comment (ShaunYearStrong) [inline public/video-ui/src/reducers/saveStateReducer.js]: To clear the `saveState` which we'll use for loading states and the like.

**Pull request merged**